### ISSUE TITLE
gocritic check-project fail when modules are enabled

### ISSUE THREAD

**Issue by mkrou** [go-critic/go-critic#651]
I made a package with testify
```go
package x   
import "github.com/stretchr/testify/assert"
```

checked by gocritic
```bash 
export GO111MODULE=on
gocritic check-project $GOPATH/src/packageWithTestify
```
got this result because gocritic doesnt know about go modules:
```bash
/home/raeven/Projects/go/pkg/mod/github.com/stretchr/testify@v1.2.2/assert/assertions.go:20:2: could not import github.com/pmezard/go-difflib/difflib (cannot find package "github.com/pmezard/go-difflib/difflib" in any of:
        /usr/local/go/src/github.com/pmezard/go-difflib/difflib (from $GOROOT)
        /home/raeven/Projects/go/src/github.com/pmezard/go-difflib/difflib (from $GOPATH))
load program: couldn't load packages due to errors: github.com/go-pg/pg, golang.org/x/crypto/bcrypt, github.com/stretchr/testify/assert and 5 more
```
without `GO111MODULE=on` gocritic works perfect

**Comment by Quasilyte** [comment]
After re-write using [golang.org/x/tools/go/packages](https://godoc.org/golang.org/x/tools/go/packages), the proper command is:
```
gocritic check packageWithTestify
```
Or:
```
gocritic check ./packageWithTestify/...
```
The `packages` package should handle modules properly.

**Comment by Quasilyte** [comment]
Note that you can also use gocritic via https://github.com/golangci/golangci-lint, which is a more user-friendly way to run checkers gocritic provides.
